### ISSUE TITLE
inequality comparison for Nat

### ISSUE THREAD

**Issue by mechvel** [agda/agda-stdlib#696]
1) I have committed a fast implementation for ``<-cmp`` in Nat.Properties.
Because the ``<-cmp`` of lib-0.17 is "exponentially" expensive in performance.

2) But  Data.Nat.Base  has ``compare`` which is "exponentially" expensive in performance,
and this ``compare`` can be used instead of  ``<-cmp`` by a confusion.
If new ``<-cmp`` is all right, then needs ``compare`` to be moved to Nat.Properties and be expressed via ``<-cmp`` ?

**Comment by MatthewDaggitt** [comment]
Yup, I guess `compare` should be deprecated in v1.1.

**Comment by oisdk** [comment]
I didn't realise `compare` was "exponential"—does someone have a link or an explanation as to this?

**Comment by mechvel** [comment]
oisdk  wrote
> I didn't realise compare was "exponential"—does someone have a link or an explanation as to this?

I meant that on performance, evaluation of the result of  ``(compare m n)`` takes time which at least exponentially depends of the value  ``(bit-length m) + (bit-length n)``. 
This can be illustrated by the following program.
```
  p : Nat =  2 ^ n;    cp = compare p (suc p);
  print the value of cp  
  (say, print (tri< _ _ _) as "less", and so on).
 ```
And let  ``n  : Nat``  be input from file. Compile this program and run it.
When  ``n``  is increased by one, the running time of this program is about doubled.
This is why I call it "exponentially" expensive.
For example, for n = 40 it will take many hours to perform on a 3 GHz machine.

And using the suggested new implementation of  ``<-cmp`` makes it cost linear rather than exponential.

**Comment by oisdk** [comment]
Thanks for the explanation!

**Comment by MatthewDaggitt** [comment]
Closed with 74e13b444dfc0f970f1ff0c21b06d9571e8312bd
